### DISCUSSION TITLE
How to create an oauth2 server by fastapi?

### DISCUSSION THREAD

**Discussion by markqiu** [fastapi/fastapi#8202] [Questions] [answered]
**Description**

How can I create an oauth2 server by fastapi?
Thank you.

**Comment by AlexandreBiguet** [comment, marked as answer]
Hi :)

I'd say that a good first step could be to check out the [fastAPI documentation](https://fastapi.tiangolo.com/tutorial/security/) where you will be able to build a fastAPI server implementing the [password grant](https://oauth.net/2/grant-types/password/)  step by step. 

If you want to use other grant type, such as the [authorisation code grant](https://oauth.net/2/grant-types/authorization-code/), I think [you can](https://github.com/tiangolo/fastapi/blob/master/fastapi/security/oauth2.py#L174), but is it not documented yet (https://github.com/tiangolo/fastapi/issues/335).

Hope this helps a bit, 
Cheers

**Comment by markqiu** [comment]
thank you very much!
